### PR TITLE
[GH-982]: Added support for webhook of plugin version v2.x.x

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
- Added the endpoint to support the webhooks created for plugin version v2.x.x

### Issue link
#982 

### What to test
-The webhook created for jira plugin version 2. x.x should be supported and working fine.
### How to test
- Create a webhook using the readme from https://github.com/mattermost/mattermost-plugin-jira/tree/v2.4.0#step-2-configure-webhooks-in-jira. 
- Create the subscription
- Change notifications should come from the jira side.

### CODE REVIEW

File: server/http.go [hunk 1]
```diff
@@ -134,6 +134,9 @@ func (p *Plugin) initializeRouter() {
 	// Firehose webhook setup for channel subscriptions
 	instanceRouter.HandleFunc(makeAPIRoute(routeAPISubscribeWebhook), p.handleResponseWithCallbackInstance(p.httpSubscribeWebhook)).Methods(http.MethodPost)
 
+	// To support Plugin v2.x webhook URL
+	apiRouter.HandleFunc(routeAPISubscribeWebhook, p.handleResponseWithCallbackInstance(p.httpSubscribeWebhook)).Methods(http.MethodPost)
```
Comment (mickmister) [inline server/http.go:138]: There are two types of webhook URLs that have the shape `/webhook` and `api/v2/webhook`. It looks like we already support the `/webhook` path with no instance ID, but not one that does contain an instance ID:

https://github.com/mattermost/mattermost-plugin-jira/blob/d1e233af901f5a4c3b652fee260857c74d1c6d44/server/http.go#L86C22-L86C42

You can run `/jira webhook` and see the section called "Legacy webhooks". That feature is essentially the original webhook implementation we had, that allows you to post to Mattermost without having to create subscriptions. Customers still use that when they don't want Mattermost to have access to their Jira server. So we still need to make sure that's covered in the `instanceRouter`

**Review by mickmister — CHANGES_REQUESTED** [review]
LGTM for the `api/v2/webhook` path, though we also need to support the `/webhook` path

File: server/http.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply server/http.go:138]: sure @mickmister looking into this.

Comment (Kshitij-Katiyar) [reply server/http.go:138]: @mickmister added additional endpoint

**Comment by mickmister** [comment]
@Kshitij-Katiyar Have you tested the following permutations for this:

- `/webhook` 
- `/api/v2/webhook`
- `instance/INSTANCE_ID/webhook`
- `instance/INSTANCE_ID/api/v2/webhook`

If so I think we can merge this

**Pull request merged**